bin/runtime/src/migrations/custom_transaction_payment_migration.rs

```diff
@@ -0,0 +1,84 @@
+use codec::{Decode, Encode, MaxEncodedLen};
+use frame_support::{
+    log,
+    pallet_prelude::{Get, TypeInfo},
+    storage_alias,
+    traits::OnRuntimeUpgrade,
+    RuntimeDebug,
+};
+use pallet_transaction_payment::{Config, NextFeeMultiplier};
+#[cfg(feature = "try-runtime")]
+use sp_std::vec::Vec;
+
+use crate::Weight;
+
+const TARGET: &str = "runtime::transaction_payment::migration";
+
+#[storage_alias]
+type StorageVersion = StorageValue<TransactionPayment, Releases>;
+
+// copied from transaction payment code as it is not pub there
+#[derive(Encode, Decode, Clone, Copy, PartialEq, Eq, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+enum Releases {
+    /// Original version of the pallet.
+    V1Ancient,
+    /// One that bumps the usage to FixedU128 from FixedI128.
+    V2,
+}
+
+/// Custom migrations the transaction payment pallet from V1 to V2 that only bumps StorageVersion to 2
+pub struct BumpTransactionVersionToV2<T>(sp_std::marker::PhantomData<T>);
+
+impl<T: Config> OnRuntimeUpgrade for BumpTransactionVersionToV2<T> {
+    fn on_runtime_upgrade() -> Weight {
+        let version = StorageVersion::get();
+        if version.is_some() && version.unwrap() == Releases::V2 {
+            log::warn!(
+                target: TARGET,
+                "skipping V1Ancient to V2 migration: executed on wrong storage version.\
+				Expected version V1Ancient, found {:?}",
```

Comment:
We can be confident here (since we just checked it in the `if`) that the `version` is `Releases::V2`.

Reply:
Right, done